src/core/gltf2importer/meshparser.cpp

```diff
@@ -123,6 +210,284 @@ Qt3DRender::QAttribute *decodeAttribute(const draco::PointCloud *pointCould,
     return attribute;
 }
 #endif
+
+int vertexBaseTypeSize(Qt3DRender::QAttribute::VertexBaseType vertexBaseType)
+{
+    switch (vertexBaseType) {
+    case Qt3DRender::QAttribute::UnsignedByte:
+        return 1;
+    case Qt3DRender::QAttribute::UnsignedShort:
+        return 2;
+    case Qt3DRender::QAttribute::UnsignedInt:
+        return 4;
+    case Qt3DRender::QAttribute::Float:
+        return 4;
+    default:
+        Q_UNREACHABLE();
+    }
+}
+
+struct Attribute {
+    QByteArray bufferData;
+    unsigned int byteOffset;
+    unsigned int byteStride;
+};
+
+// Chances to visit the same face again are high, so cache the state of the last search
+struct FindVertexIndicesInFaceHelper {
+
+    FindVertexIndicesInFaceHelper()
+        : lastFace(-1)
+    {
+    }
+
+    const std::array<unsigned int, 3> &operator()(int iFace) const
+    {
+        if (iFace == lastFace)
+            return verticesForLastFace;
+
+        lastFace = iFace;
+        const auto data = indexAttribute.bufferData;
+        const auto iFaceHead = data.mid(indexAttribute.byteOffset + iFace * indexAttribute.byteStride);
+        switch (vertexBaseType) {
+        case Qt3DRender::QAttribute::UnsignedByte: {
+            const auto *typedIndices = reinterpret_cast<const unsigned char *>(iFaceHead.data());
+            verticesForLastFace = { typedIndices[0], typedIndices[1], typedIndices[2] };
+            break;
+        }
+        case Qt3DRender::QAttribute::UnsignedShort: {
+            const auto *typedIndices = reinterpret_cast<const unsigned short *>(iFaceHead.data());
+            verticesForLastFace = { typedIndices[0], typedIndices[1], typedIndices[2] };
+            break;
+        }
+        case Qt3DRender::QAttribute::UnsignedInt: {
+            const auto *typedIndices = reinterpret_cast<const unsigned int *>(iFaceHead.data());
+            verticesForLastFace = { typedIndices[0], typedIndices[1], typedIndices[2] };
+            break;
+        }
+        default:
+            Q_UNREACHABLE();
+        }
+        return verticesForLastFace;
+    }
+
+    Attribute indexAttribute;
+    Qt3DRender::QAttribute::VertexBaseType vertexBaseType;
+    mutable int lastFace;
+    mutable std::array<unsigned int, 3> verticesForLastFace;
+};
+
+struct MikkTSpaceUserData {
```

Comment:
clang reports warnings about adding padding to keep memory alignment to members of this struct. Moving `primitiveType` to before `vertexIndicesFinder` fixes those.

This (along with a lot of the pre-existing code in this file) produces warnings of signed/unsigned conversion in places that mix int and uint. `nFaces` and `nVertices` should be uint here I think